base/pmap.jl

```diff
@@ -93,9 +75,8 @@ function batchsplit(c; min_batch_count=1, max_batch_size=100)
 
     # If there are not enough batches, use a smaller batch size...
     if length(head) < min_batch_count
-        head = vcat(head...)
-        batch_size = max(1, div(length(head), min_batch_count))
-        return partition(head, batch_size)
+        batch_size = max(1, div(sum(length, head), min_batch_count))
+        return partition(flatten(head), batch_size)
     end
```

Comment:
@amitmurthy with reference to [our previous `vcat` discussion](https://github.com/JuliaLang/julia/pull/15409/files/b845ea1ec5450003f1f749013ae482faac8d7329#r57449634), I'm now using `flatten` instead of `vcat` in `batchsplit` because `vcat` doesn't do the expected thing with strings. (See also https://github.com/JuliaLang/julia/pull/14383#issuecomment-202596418)